qemu/tests/balloon_check.py

```diff
@@ -148,6 +148,19 @@ def run_balloon_sub_test(self, test, params, env, test_tag):
             qemu_quit_after_test = 0
         return qemu_quit_after_test
 
+    def wait_for_balloon_complete(self):
+        """
+        Wait until guest memory don't change after do balloon
+        """
+        logging.info("Wait until guest memory don't change")
+        gmem = self.get_memory_status()
+        while True:
+            time.sleep(30)
+            cur_gmem = self.get_memory_status()
+            if abs(cur_gmem - gmem) < 100:
+                break
+            gmem = cur_gmem
```

Comment:
I don't think utils_misc.wait_for() fits here.  
`def wait_for(func, timeout, first=0.0, step=1.0, text=None)`
The func() takes no parameters . Here we need to compare the memory every 'step' time. I didn't see a way to achieve this unless we add a new class variable to record the memory every time. Do you thinks it's a good idea, or it costs too much? Anyway, it is a good suggestion to set a timeout and I have update it to current code. Thanks!